lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -271,6 +276,17 @@ Status ProcessElfCore::DoLoadCore() {
   return error;
 }
 
+void ProcessElfCore::UpdateBuildIdForNTFileEntries() {
+  if (!m_nt_file_entries.empty()) {
+    for (NT_FILE_Entry &entry : m_nt_file_entries) {
+      std::optional<UUID> uuid =
```

Comment:
Since UUID already has an IsValid() method I don't think wrapping it in an optional is not going to help (in fact, it will add ambiguity: Does returning an invalid UUID mean something different than std::nullopt?) Up to you, but I'd remove the optional here...